Assets/HoloToolkit/UI/Scripts/Buttons/CompoundButtonIcon.cs

```diff
@@ -0,0 +1,313 @@
+        /// </summary>
+        public void OnWillSaveScene()
+        {
+            ClearInstancedAssets();
+
+            SetIconName(iconName);
+            
+        }
+        #endif
+
+        public string IconName
+        {
+            get
+            {
+                return iconName;
+            }
+            set
+            {
+                SetIconName(value);
+            }
+        }
+
+        private void SetIconName(string newName)
+        {
+            // Avoid exploding if possible
+            if (Profile == null)
+                return;
+
+            if (IconRenderer == null)
+                return;
+
+            if (DisableIcon)
+            {
+                IconRenderer.enabled = false;
+                return;
+            }
+
+            if (Profile.IconMaterial == null || Profile.IconMesh == null)
+                return;
+
+            // Instantiate our local material now, if we don't have one
+            if (instantiatedMaterial == null)
+            {
+                instantiatedMaterial = new Material(Profile.IconMaterial);
+                instantiatedMaterial.name = Profile.IconMaterial.name;
+            }
+            IconRenderer.sharedMaterial = instantiatedMaterial;
+            
+            // Instantiate our local mesh now, if we don't have one
+            if (instantiatedMesh == null)
+            {
+                instantiatedMesh = Mesh.Instantiate (Profile.IconMesh) as Mesh;
+                instantiatedMesh.name = Profile.IconMesh.name;
+            }
+            IconMeshFilter.sharedMesh = instantiatedMesh;
+
+            if (OverrideIcon)
+            {
+                // Use the default mesh for override icons
+                IconRenderer.enabled = true;
+                IconMeshFilter.sharedMesh = Profile.IconMesh;
+                IconMeshFilter.transform.localScale = Vector3.one;
+                instantiatedMaterial.mainTexture = iconOverride;
+                return;
+            }
+
+            // Disable the renderer if the name is empty
+            if (string.IsNullOrEmpty(newName))
+            {
+                IconRenderer.enabled = false;
+                iconName = newName;
+                return;
+            }
+            
+            // Moment of truth - try to get our icon
+            if (!Profile.GetIcon(iconName, IconRenderer, IconMeshFilter, true))
+            {
+                IconRenderer.enabled = false;
+                return;
+            }
+
+            // If we've made it this far we're golden
+            IconRenderer.enabled = true;
+            iconName = newName;
+            RefreshAlpha();
+        }
+
+        private void OnDisable()
+        {
+            ClearInstancedAssets();
+        }
+
+        private void OnEnable()
+        {
+            if (Application.isPlaying)
+            {
+                ClearInstancedAssets();
+            }
+
+            SetIconName(iconName);
```

Comment:
I've had issues with this code. When enabling/disabling buttons several times the uv-rects of the icons start breaking. See an example here: https://puu.sh/xIL06/0c20ded28c.webm.

This ended up solving it for me but it's more of a workaround than an actual fix, maybe someone with more time or knowledge of the code can figure it out:
```csharp
private void OnEnable()
{
    if (Application.isPlaying)
    {
        ClearInstancedAssets();
    }

    StartCoroutine(RefreshIcon());
}

private IEnumerator RefreshIcon()
{
    SetIconName(iconName);
    yield return new WaitForEndOfFrame();
    SetIconName(iconName); // UVRect of the icon atlas will be broken without this, but why??
}
```
Also a tip for debugging it: You can trigger the fix by selecting the button object in the hierarchy because that triggers   setting the icon again from the property drawer.

Reply:
@Zod- Thanks for pointing this out.  Yeah, this looks like some bad ref issues.  I'll look into it.  I didn't originally write the code and I know we have a thread about the issue on the MRDL GitHub repo.

Reply:
@Zod- I have experienced similar issue. I'll ask Lars who created this.
Thanks!

Reply:
This is unfortunately due to a [common font glitch in Unity](https://www.google.com/images?q=unity+garbled+fonts) which we've attempted to mitigate (with limited success). Some causes are fairly deterministic but others are mysterious.

We've got an [issue on it here](https://github.com/Microsoft/MRDesignLabs_Unity/issues/64) with @Zod's fix.

Surprisingly I've yet to see this in [Unity's issue tracker.](https://unity3d.com/search?refinement=issues&gq=font) We should try to get a bug submitted.